File: Android/app/src/main/java/kalen/app/ustb_choose_course_system/utils/HtmlParser.java
package kalen.app.ustb_choose_course_system.utils;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.util.ArrayList;
import java.util.List;

import kalen.app.ustb_choose_course_system.model.ExamInfo;
import kalen.app.ustb_choose_course_system.model.InnovateCredit;

/**
 * Created by kalen on 15-11-24.
 */
public class HtmlParser {
    public static List<InnovateCredit> getInnovateCredits(String html){
        List<InnovateCredit> credits = new ArrayList<>();

        Document doc = Jsoup.parse(html);
        Elements elements = doc.select("tbody tr");
        for (Element element : elements){
            Elements elems = element.select("td");
            String type = elems.get(1).text();
            String name = elems.get(2).text();
            String credit = elems.get(3).text();

            InnovateCredit credit1 = new InnovateCredit(credit, name, type);
            credits.add(credit1);
        }

        return credits;
    }

    public static List<ExamInfo> getExamLists(String html){
        List<ExamInfo> examInfos = new ArrayList<>();

        Document doc = Jsoup.parse(html);
        Elements elements = doc.select("tbody tr");
        for (Element element : elements){
            Elements elems = element.select("td");
            String className = elems.get(1).text();
            String examTime = elems.get(2).text();
            String examLocation = elems.get(3).text();
            ExamInfo info = new ExamInfo(className, examTime, examLocation);
            examInfos.add(info);
        }

        return examInfos;
    }

   // public static

}

File: iOS/HtmlParser.swift
//
//  HtmlParser.swift
//  ustb_choose_course_system
//
//  Created by kalen blue on 16/1/14.
//  Copyright © 2016年 kalen blue. All rights reserved.
//

import UIKit

extension kalen.app{
    class HtmlParser: NSObject {
        
        class func getInnovateCredits(html:String)->[InnovativeCredit]{
            var credits :[InnovativeCredit] = []
            let doc = try? HTMLDocument(string: html, encoding: NSUTF8StringEncoding)
            var index = 0
            for tr in doc!.css("tbody, tr") {
                index += 1
                let tds = tr.css("td")
                var i = 0
                let credit = InnovativeCredit(credit: "", name: "", type: "")
                for td in tds{
                    td.stringValue
                    switch i{
                    case 1:credit.type = td.stringValue
                    case 2:credit.name = td.stringValue
                    case 3:credit.credit = td.stringValue
                    default: break
                    }
                    i += 1
                }
                print(index,credit.name)
                credits.append(credit)
            }
            credits.removeFirst()
            credits.removeFirst()
            return credits
        }
        
        class func getExamList(html:String)->[ExamInfo]{
            var list :[ExamInfo] = []
            let doc = try? HTMLDocument(string: html, encoding: NSUTF8StringEncoding)
            for tr in doc!.css("tbody, tr") {
                let tds = tr.css("td")
                var i = 0
                let exam = ExamInfo(className: "", examTime: "", examLocation: "")
                for td in tds{
                    td.stringValue
                    switch i{
                    case 1:exam.className = td.stringValue
                    case 2:exam.examTime = td.stringValue
                    case 3:exam.examLocation = td.stringValue
                    default: break
                    }
                    i += 1
                }
                list.append(exam)
            }
            list.removeFirst()
            list.removeFirst()
            return list
        }
        
        
    }
}

